### DISCUSSION TITLE
Failed sending data to the peer (Connection died, tried 5 times before giving up)

### DISCUSSION THREAD

**Discussion by jflessau** [community/community#25687] [Actions]
<p>When <code>cargo check</code> runs in a github action it downloads most of the dependencies, but occasionally fails on one of them. It fails on different ones on each run (tested it around 5 times).</p>
<p>Downloading dependencies works locally, but when running in a github action it fails:</p>
<pre><code class="lang-auto">Downloaded sys-info v0.5.10
  Downloaded strum_macros v0.19.4
  Downloaded proc-macro-error v1.0.4
  Downloaded pin-utils v0.1.0
  Downloaded v_escape v0.15.0
  Downloaded uuid v0.7.4
  Downloaded typeable v0.1.2
  Downloaded syn v1.0.73
  Downloaded strum v0.20.0
error: failed to download from `https://crates.io/api/v1/crates/rustc-demangle/0.1.20/download`
Error: failed to download from `https://crates.io/api/v1/crates/rustc-demangle/0.1.20/download`
Caused by:
  [55] Failed sending data to the peer (Connection died, tried 5 times before giving up)
Error: The process '/usr/share/rust/.cargo/bin/cargo' failed with exit code 101
</code></pre>
<p>I really don’t know what’s causing this. Does anyone have a hint for me?</p>
<p>Thank you in advance :slight_smile:</p>

**Comment by vadixidav** [comment]
<p>I am hitting similar issues over here <a href="https://github.com/rust-cv/cv/pull/33/checks?check_run_id=3038828890" class="inline-onebox" rel="noopener nofollow ugc">Update nalgebra and the rest of dependencies to have the crates compile by mpizenberg · Pull Request #33 · rust-cv/cv · GitHub</a>. I am wondering if there are network limits we are running into? I am getting failed dependency downloads. To mitigate this, I am trying to use <a href="https://github.com/marketplace/actions/rust-cache" rel="noopener nofollow ugc">rust-cache</a> to cache some of the source so it doesn’t need to be redownloaded. If anyone knows of a proper fix, that would be awesome.</p>

**Comment by luigi311** [comment]
<p>I am also running into this often requiring the entire workflow to be restarted</p>

**Comment by jflessau** [comment, marked as answer]
<p>Hi there!<br>
A colleague found a solution for this. Seems like it’s a problem with nightly. This change in our ci file fixed it:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/7/e/7e755f1372f76664af1561c556caadf675c76a2c.png" data-download-href="/uploads/short-url/i2HCXgur2tyq4LUkYA0JgeaBuTG.png?dl=1" title="Bildschirmfoto 2021-07-12 um 15.58.43" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/20475387/181092505-77419fca-b1e9-4b33-8949-cd7b41a87530.png" alt="Bildschirmfoto 2021-07-12 um 15.58.43" data-base62-sha1="i2HCXgur2tyq4LUkYA0JgeaBuTG" width="345" height="142" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/7/e/7e755f1372f76664af1561c556caadf675c76a2c_2_345x142.png, https://aws1.discourse-cdn.com/github/optimized/3X/7/e/7e755f1372f76664af1561c556caadf675c76a2c_2_517x213.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/7/e/7e755f1372f76664af1561c556caadf675c76a2c_2_690x284.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/7/e/7e755f1372f76664af1561c556caadf675c76a2c_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Bildschirmfoto 2021-07-12 um 15.58.43</span><span class="informations">720×298 22.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by mars64** [comment]
<p>Version locking to <code>nightly-2021-07-05</code> seems to have resolved this error for me as well. However, this doesn’t appear to be limited to GitHub Actions.</p>
<p>I’m running <code>cargo +nightly fmt --all -- --check</code> in a GitLab-CI custom runner on an EKS cluster with NAT gateway egress.  Sometimes it works, sometimes it errors with <code>Error during execution of cargo metadata: Updating git repository</code>, then proceeds to reinstall a bunch of packages eventually bombing out with the same error cause <code>[55]</code>.</p>
<p>Thanks for the tip.</p>

**Comment by vadixidav** [comment]
<p>Is anyone aware if there is a tracking issue for this problem if it is a bug in Cargo?</p>
